### PR TITLE
[AFD] Introduce and use pool tags. CORE-14048

### PR DESCRIPTION
**Pull request by AmineKhaldi**
Thanks to @prozacchiwawa for assisting me with this.

### CODE REVIEW

File: drivers/network/afd/afd/write.c [hunk 1]
```diff
@@ -398,7 +398,7 @@ AfdConnectedSocketWriteData(PDEVICE_OBJECT DeviceObject, PIRP Irp,
                 }
             }
 
-            ExFreePool( TargetAddress );
+            ExFreePoolWithTag(TargetAddress, TAG_AFD_TRANSPORT_ADDRESS);
```
Comment (ThFabba) [inline drivers/network/afd/afd/write.c]: TAG_AFD_TDI_CONNECTION_INFORMATION

File: drivers/network/afd/afd/lock.c [hunk 2]
```diff
@@ -69,7 +69,10 @@ PVOID LockRequest( PIRP Irp,
                 }
 
                 /* The allocated address goes in index 0 */
-                Irp->Tail.Overlay.DriverContext[0] = ExAllocatePool(NonPagedPool, MmGetMdlByteCount(Irp->MdlAddress));
+                Irp->Tail.Overlay.DriverContext[0] = ExAllocatePoolWithTag(NonPagedPool,
```
Comment (ThFabba) [inline drivers/network/afd/afd/lock.c]: This (and AfdInfo below) could have its own tag; there's not really a connection to Recv.Window/Send.Window, right?

File: drivers/network/afd/afd/tdi.c [hunk 3]
```diff
@@ -229,13 +231,16 @@ NTSTATUS TdiQueryMaxDatagramLength(
     PTDI_MAX_DATAGRAM_INFO Buffer;
     NTSTATUS Status = STATUS_SUCCESS;
 
-    Buffer = ExAllocatePool(NonPagedPool, sizeof(TDI_MAX_DATAGRAM_INFO));
+    Buffer = ExAllocatePoolWithTag(NonPagedPool,
```
Comment (ThFabba) [inline drivers/network/afd/afd/tdi.c]: I guess this one could also have its own tag. But I guess you have enough tags, so fair enough :p

File: drivers/network/afd/afd/write.c (hunk 1 above)
Comment (AmineKhaldi) [reply drivers/network/afd/afd/write.c]: And I thought I was careful.. Thanks! :)

File: drivers/network/afd/afd/lock.c (hunk 2 above)
Comment (AmineKhaldi) [reply drivers/network/afd/afd/lock.c]: Sure, what do you have in mind?

File: drivers/network/afd/afd/tdi.c (hunk 3 above)
Comment (AmineKhaldi) [reply drivers/network/afd/afd/tdi.c]: We do indeed have so many tags, but we're not running out of them anytime soon are we? :)
If you would like to see this have its own tag I'm all for it.

Comment (ThFabba) [reply drivers/network/afd/afd/tdi.c]: Nah, this seems fine for an initial run through. If someone finds more leaks later maybe they'll refine it more.

**Pull request merged**